src/App.tsx

```diff
@@ -15,35 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = '',
+  SORT_FIELD_ALPHABET = 'alphabet',
+  SORT_FIELD_LENGTH = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: { sortField: SortType; isReversed: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goodsOnSite = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const reset = () => {
+    setSortField(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SortType.SORT_FIELD_ALPHABET,
+          })}
+          onClick={() => setSortField(SortType.SORT_FIELD_ALPHABET)}
+        >
           Sort alphabetically
```

Comment:
[UX] - To improve user experience, consider marking the active button more clearly to indicate the current sorting applied. For instance, you could change the button style or add an icon to show the active state.